drivers/SmartThings/matter-appliance/src/matter-laundry/init.lua

```diff
@@ -100,6 +143,84 @@ local function supported_temperature_levels_attr_handler(driver, device, ib, res
+    return
+  end
+  device.log.info(string.format("temperature_setpoint_attr_handler: %d", ib.data.value))
+
+  local min_field = string.format("%s-%d", setpoint_limit_device_field.MIN_TEMP, ib.endpoint_id)
+  local max_field = string.format("%s-%d", setpoint_limit_device_field.MAX_TEMP, ib.endpoint_id)
+  local min, max
+  local laundry_device_type = device:get_field(LAUNDRY_DEVICE_TYPE_ID)
+  if laundry_device_type == LAUNDRY_DRYER_DEVICE_TYPE_ID then
+    min = device:get_field(min_field) or LAUNDRYDRYER_MIN_TEMP_IN_C
+    max = device:get_field(max_field) or LAUNDRYDRYER_MAX_TEMP_IN_C
+  elseif laundry_device_type == LAUNDRY_WASHER_DEVICE_TYPE_ID then
+    min = device:get_field(min_field) or LAUNDRYWASHER_MIN_TEMP_IN_C
+    max = device:get_field(max_field) or LAUNDRYWASHER_MAX_TEMP_IN_C
+  else
+    device.log.warn(string.format("Not a supported device type"))
+    return
+  end
+  local temp = ib.data.value / 100.0
+  local unit = "C"
+  local range = {
+    minimum = min,
+    maximum = max,
+  }
+
+  -- Only emit the capability for RPC version >= 5, since unit conversion for
+  -- range capabilities is only supported in that case.
+  if version.rpc >= 5 then
+    device:emit_event_for_endpoint(ib.endpoint_id,
+      capabilities.temperatureSetpoint.temperatureSetpointRange({value = range, unit = unit}, {visibility = {displayed = false}}))
+  end
+
+  device:emit_event_for_endpoint(ib.endpoint_id,
+    capabilities.temperatureSetpoint.temperatureSetpoint({value = temp, unit = unit}))
+end
+
+local function setpoint_limit_handler(limit_field)
+  return function(driver, device, ib, response)
+    local tn_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID,
+      { feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_NUMBER })
+    if #tn_eps == 0 then
+      device.log.warn(string.format("Device does not support TEMPERATURE_NUMBER feature"))
+      return
+    end
+    local field = string.format("%s-%d", limit_field, ib.endpoint_id)
+    local val = ib.data.value / 100.0
+
+    local min_temp_in_c, max_temp_in_c
+    local laundry_device_type = device:get_field(LAUNDRY_DEVICE_TYPE_ID)
+    if laundry_device_type == LAUNDRY_DRYER_DEVICE_TYPE_ID then
+      min_temp_in_c = LAUNDRYDRYER_MIN_TEMP_IN_C
+      max_temp_in_c = LAUNDRYDRYER_MAX_TEMP_IN_C
+    elseif laundry_device_type == LAUNDRY_WASHER_DEVICE_TYPE_ID then
+      min_temp_in_c =  LAUNDRYWASHER_MIN_TEMP_IN_C
+      max_temp_in_c =  LAUNDRYWASHER_MAX_TEMP_IN_C
+    else
+      device.log.warn(string.format("Not a supported device type"))
+      return
+    end
+    -- We clamp the max and min values as per the assumed oven temperature ranges.
+    if val < min_temp_in_c or val > max_temp_in_c then
+      if limit_field == setpoint_limit_device_field.MIN_TEMP then
+        val = min_temp_in_c
+      else
+        val = max_temp_in_c
+      end
+    end
+
+    log.info("Setting " .. field .. " to " .. string.format("%s", val))
```

Comment:
I would use device.log so that the device info is automatically prepended to the log
```suggestion
    device.log.info("Setting " .. field .. " to " .. string.format("%s", val))
```